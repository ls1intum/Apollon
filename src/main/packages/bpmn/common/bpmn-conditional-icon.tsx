import React, { FunctionComponent } from 'react';
import { ThemedPolyline, ThemedRect } from '../../../components/theme/themedComponents';

export const BPMNConditionalIcon: FunctionComponent<React.SVGProps<SVGSVGElement>> = (props) => (
  <svg {...props} height={20} width={20}>
    <ThemedRect height={16} width={16} x={2} y={2} strokeLinejoin="round" />
    <ThemedPolyline points={'6 7, 14 7'} strokeLinecap="round" strokeLinejoin="round" />
    <ThemedPolyline points={'6 13, 14 13'} strokeLinecap="round" strokeLinejoin="round" />
  </svg>
);
