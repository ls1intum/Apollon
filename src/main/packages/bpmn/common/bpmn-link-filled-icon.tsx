import React, { FunctionComponent } from 'react';
import { ThemedPolyline } from '../../../components/theme/themedComponents';

export const BPMNLinkFilledIcon: FunctionComponent<React.SVGProps<SVGSVGElement>> = (props) => (
  <svg {...props} height={20} width={20}>
    <ThemedPolyline
      points={'3 7, 13 7, 13 4, 18 10, 13 16, 13 13, 3 13, 3 7'}
      strokeLinecap="round"
      strokeLinejoin="round"
      fillColor="currentColor"
      fillRule="evenodd"
    />
  </svg>
);
