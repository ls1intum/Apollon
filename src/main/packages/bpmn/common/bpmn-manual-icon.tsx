import React, { FunctionComponent } from 'react';
import { ThemedPath } from '../../../components/theme/themedComponents';

export const BPMNManualIcon: FunctionComponent<React.SVGProps<SVGSVGElement>> = (props) => (
  <svg {...props} height={20} width={20}>
    <ThemedPath
      d="M5 2.00001C5 3.00001 5 10 5 10M5 2.00001C5 1 7 0.999998 7 1.99999M5 2.00001C5 2.00001 5 5 5 4C5 2.99999 3 2.93337 3 4C3 5.06663 3 12 3 12C3 12 3 11 3 10C3 8.99999 1 8.99999 1 10C1 11 1 14 1 14L3 16V19L9 19V16L11 14V10.5625C11 10.5625 11 7 11 6C11 5 9 5 9 6C9 7 9 4 9 4M7 1.99999C7 2.99998 7 10 7 10M7 1.99999C7 1.99999 7 5 7 4C7 3 9 3 9 4M9 4V10"
      strokeLinejoin="round"
    />
  </svg>
);
