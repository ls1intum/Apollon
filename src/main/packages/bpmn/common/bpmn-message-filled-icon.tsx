import React, { FunctionComponent } from 'react';
import { ThemedPolyline } from '../../../components/theme/themedComponents';

export const BPMNMessageFilledIcon: FunctionComponent<React.SVGProps<SVGSVGElement>> = (props) => (
  <svg {...props} height={20} width={20}>
    <ThemedPolyline
      points={'0.2 3, 19.8 3, 10 11, 0.2 3'}
      strokeLinecap="round"
      strokeLinejoin="round"
      fillColor="currentColor"
    />
    <ThemedPolyline
      points={'0 5.5, 0 17, 20 17, 20 5.5, 10 13.5, 0 5.5'}
      strokeLinecap="round"
      strokeLinejoin="round"
      fillColor="currentColor"
    />
  </svg>
);
