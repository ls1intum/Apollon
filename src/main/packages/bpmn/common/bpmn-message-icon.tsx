import React, { FunctionComponent } from 'react';
import { ThemedPolyline } from '../../../components/theme/themedComponents';

export const BPMNMessageIcon: FunctionComponent<React.SVGProps<SVGSVGElement>> = (props) => (
  <svg {...props} height={20} width={20}>
    <ThemedPolyline points={'0 3, 0 17, 20 17, 20 3, 10 11, 0 3, 20 3'} strokeLinecap="round" strokeLinejoin="round" />
  </svg>
);
