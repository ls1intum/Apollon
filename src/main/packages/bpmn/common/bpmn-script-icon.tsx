import React, { FunctionComponent } from 'react';
import { ThemedPolyline, ThemedRect } from '../../../components/theme/themedComponents';

export const BPMNScriptIcon: FunctionComponent<React.SVGProps<SVGSVGElement>> = (props) => (
  <svg {...props} height={20} width={20}>
    <ThemedRect height={16} width={16} x={2} y={2} strokeLinejoin="round" />
    <ThemedPolyline points={'6 6, 12 6'} strokeLinecap="round" strokeLinejoin="round" />
    <ThemedPolyline points={'6 10, 14 10'} strokeLinecap="round" strokeLinejoin="round" />
    <ThemedPolyline points={'6 14, 10 14'} strokeLinecap="round" strokeLinejoin="round" />
  </svg>
);
