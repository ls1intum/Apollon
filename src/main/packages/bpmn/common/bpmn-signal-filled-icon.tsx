import React, { FunctionComponent } from 'react';
import { ThemedPolyline } from '../../../components/theme/themedComponents';

export const BPMNSignalFilledIcon: FunctionComponent<React.SVGProps<SVGSVGElement>> = (props) => (
  <svg {...props} height={20} width={20}>
    <ThemedPolyline
      points={'10 3, 3 15, 17 15, 10 3'}
      strokeLinecap="round"
      strokeLinejoin="round"
      fillColor="currentColor"
      fillRule="evenodd"
    />
  </svg>
);
