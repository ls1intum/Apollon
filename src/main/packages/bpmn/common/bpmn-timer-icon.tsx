import React, { FunctionComponent } from 'react';
import { ThemedCircle, ThemedPolyline } from '../../../components/theme/themedComponents';

export const BPMNTimerIcon: FunctionComponent<React.SVGProps<SVGSVGElement>> = (props) => (
  <svg {...props} height={20} width={20}>
    <ThemedCircle cx="50%" cy="50%" r={10} />
    <ThemedPolyline points={'10 4, 10 10, 13 13'} strokeLinecap="round" strokeLinejoin="round" />
  </svg>
);
