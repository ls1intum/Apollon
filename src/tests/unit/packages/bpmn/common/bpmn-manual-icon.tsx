import { wrappedRender } from '../../../test-utils/render';
import * as React from 'react';
import { BPMNManualIcon } from '../../../../../main/packages/bpmn/common/bpmn-manual-icon';

it('render the bpmn-manual-icon', () => {
  const { baseElement } = wrappedRender(
    <svg>
      <BPMNManualIcon />
    </svg>,
  );
  expect(baseElement).toMatchSnapshot();
});
