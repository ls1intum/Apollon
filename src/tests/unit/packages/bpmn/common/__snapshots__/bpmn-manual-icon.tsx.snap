// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the bpmn-manual-icon 1`] = `
<body>
  <div>
    <svg>
      <svg
        height="20"
        width="20"
      >
        <path
          class="sc-gEvEer JpXgg"
          d="M5 1.5C5 2.5 5 8 5 8M5 1.5C5 0.499993 7 0.500007 7 1.5M5 1.5C5 1.5 5 4.00001 5 3.00001C5 2 3 1.93337 3 3.00001C3 4.06664 3 10 3 10C3 10 3 8.25001 3 7.25001C3 6.25 1 6.25 1 7.25001C1 8.25001 1 12 1 12L3 14V16H9V14L11 12V8.56252C11 8.56252 11 5 11 4C11 3 9 3 9 4C9 5 9 3.00001 9 3.00001M7 1.5C7 2.49999 7 8 7 8M7 1.5C7 1.5 7 4.00001 7 3.00001C7 2.00001 9 2.00001 9 3.00001M9 3.00001V8"
          fill="transparent"
          stroke="black"
          stroke-linejoin="round"
        />
      </svg>
    </svg>
  </div>
</body>
`;
